### PR TITLE
For #9184 - Emit ContextMenuFact to track text selection context usage in Fenix

### PR DESCRIPTION
**Pull request by codrut-topliceanu**
For https://github.com/mozilla-mobile/android-components/issues/9184

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks.
- [x] **Tests**: This PR does not include any tests as it is only a minor change.
- [x] **Changelog**: This PR doesn't include [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) as this is only a minor change.
- [x] **Accessibility**: The code in this PR does not include any user facing features.

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Comment by jonalmeida** [comment]
For telemetry, we emit "[facts](https://github.com/mozilla-mobile/android-components/blob/master/components/support/base/src/main/java/mozilla/components/support/base/facts/Fact.kt)" for consumers to track.

You can see some examples of facts being emitted in other components [here](https://searchfox.org/mozilla-mobile/search?q=emit%5Ba-zA-Z%5D%2BFact&regexp=true&path=).

File: components/feature/contextmenu/src/main/java/mozilla/components/feature/contextmenu/DefaultSelectionActionDelegate.kt [hunk 1]
```diff
@@ -68,28 +69,31 @@ class DefaultSelectionActionDelegate(
         else -> null
     }
 
-    override fun performAction(id: String, selectedText: String): Boolean = when (id) {
-        SEARCH -> {
-            searchAdapter.sendSearch(false, selectedText)
-            true
+    override fun performAction(id: String, selectedText: String): Boolean {
+        trackContextMenuOption?.invoke(id)
```
Comment (csadilek) [inline components/feature/contextmenu/src/main/java/mozilla/components/feature/contextmenu/DefaultSelectionActionDelegate.kt:73]: Just a drive-by: Please add a unit test to `DefaultSelectionActionDelegateTest` to make sure we're invoking the lambda.

Comment (codrut-topliceanu) [reply components/feature/contextmenu/src/main/java/mozilla/components/feature/contextmenu/DefaultSelectionActionDelegate.kt]: Hey, thanks for taking a look. I've switched to emitFact method that @jonalmeida suggested.

File: components/feature/contextmenu/src/main/java/mozilla/components/feature/contextmenu/DefaultSelectionActionDelegate.kt [hunk 2]
```diff
@@ -68,28 +69,31 @@ class DefaultSelectionActionDelegate(
         else -> null
     }
 
-    override fun performAction(id: String, selectedText: String): Boolean = when (id) {
-        SEARCH -> {
-            searchAdapter.sendSearch(false, selectedText)
-            true
+    override fun performAction(id: String, selectedText: String): Boolean {
+        emitTextSelectionClickFact(id)
```
Comment (jonalmeida) [inline components/feature/contextmenu/src/main/java/mozilla/components/feature/contextmenu/DefaultSelectionActionDelegate.kt:73]: Looks good! Let's add a simple test that checks `performAction` always emits a fact so we don't mistakenly remove this call.

**Comment by codrut-topliceanu** [comment]
@jonalmeida Hey, I've switched over to facts, could you or @csadilek take a look and tell me if it's okay? :)

**Review by jonalmeida — APPROVED** [review]
LGTM! Thanks for the ping!
